gamedata/alldefs_post.lua

```diff
@@ -238,8 +238,41 @@ function UnitDef_Post(name, uDef)
 			end
 		end
 
+		--normal commander respawning
+		if modOptions.comrespawn == "all" then
+			if name == "armcom" or name == "corcom" or name == "legcom" then
+				uDef.customparams.effigy = "comeffigylvl1"
+				uDef.customparams.respawn_condition = "health"
+				uDef.customparams.respawn_announcement = "A Commander Effigy was sacrificed."
+				uDef.customparams.respawn_announcement_size = 18.5
+				uDef.customparams.minimum_respawn_stun = 5
+				uDef.customparams.distance_stun_multiplier = 1
+				local numBuildoptions = #uDef.buildoptions
+				uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl1"
+			end
+		end
+		
+
 		if modOptions.evocom then	
 			if uDef.customparams.isevocom or name == "armcom" or name == "corcom" or name == "legcom" then
+				
+				if modOptions.comrespawn == "all" or modOptions.comrespawn == "evocom" then--add effigy respawning, if enabled
+					uDef.customparams.respawn_condition = "health"
+					
+					Spring.Echo("flarf 1")
+					local numBuildoptions = #uDef.buildoptions
+					if string.sub(name, -3) == "com" or string.sub(name, -7) == "comlvl2" then
+						uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl1"
+					elseif string.sub(name, -7) == "comlvl3" or string.sub(name, -7) == "comlvl4" then
+						uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl2"
+					elseif string.sub(name, -7) == "comlvl5" or string.sub(name, -7) == "comlvl6" then
+						uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl3"
+					elseif string.sub(name, -7) == "comlvl7" or string.sub(name, -7) == "comlvl8" then
+						uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl4"
+					elseif string.sub(name, -8) == "comlvl10" or string.sub(name, -7) == "comlvl9" then
+						uDef.buildoptions[numBuildoptions + 1] = "comeffigylvl5"
+					end
```

Comment:
String parsing is extremely fragile and buggy, use properties on the unitdef instead

Reply:
An additional customparam perhaps that lists the commander's level? OK.

Reply:
![image](https://github.com/beyond-all-reason/Beyond-All-Reason/assets/165520713/6038e99e-014d-4dae-996b-5c09f7df5052)

Also, is this a necessary change when these string.sub functions are only ran on units with the customparam isevocom?

Reply:
The name of the effigy to spawn should be a custom param on the unit def; or the level a custom param on the Commander and effigy unit defs, and a function correlates Commander level with effigy level to spawn.

Reply:
Should've done this to begin with. I reworked all the unitdefs and the alldefs in accordance with your suggestion.